afl-cmin.py

```diff
@@ -312,6 +329,22 @@ def main():
             already_have.add(t)
         count += 1
 
+    if args.crash_dir:
+        logger.info('Saving crashes to %s', args.crash_dir)
+        crash_files = [files[c] for c in crashes]
+        crash_files, hashmap = dedup(crash_files)
+        for crash_path in crash_files:
+            fn = base64.b16encode(hashmap[crash_path]).lower()
+            output_path = os.path.join(args.crash_dir, fn)
+            try:
+                os.link(crash_path, output_path)
+            except OSError:
+                try:
+                    shutil.copy(crash_path, output_path)
+                except shutil.Error:
+                    # files are the same; do nothing since we already have it
```

Comment:
I don't understand this case. They are dedup'ed, how are they still "the same"?

Reply:
I think this can happen when files are hardlinks of one another (i.e. when they have the same inode). This happened to me several times, so I thought it be best to handle this edge case.

Reply:
I confirm that this happens when the src and dest are hardlinks of one another. I clarified the comment to reflect this.